kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ClustersUtil.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.stream.Collector;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxyspec.Clusters;
+
+public class ClustersUtil {
+    private ClustersUtil() {
+    }
+
+    static List<Clusters> distinctClusters(KafkaProxy primary) {
+        return distinctClusters(primary.getSpec().getClusters());
+    }
+
+    static List<Clusters> distinctClusters(List<Clusters> clusters) {
```

Comment:
This shouldn't strictly be necessary, because the API server should be validating the cluster uniqueness. But it was kinda simpler to add this than actually treating it as an error. It gets called multiple times during a reconciliation, which isn't ideal.

Reply:
Why not throw an exception here if it's a sanity check?